lib/IRGen/GenMeta.cpp

```diff
@@ -4967,6 +4973,36 @@ void irgen::emitClassMetadata(IRGenModule &IGM, ClassDecl *classDecl,
   }
 }
 
+void irgen::emitEmbeddedClassMetadata(IRGenModule &IGM, ClassDecl *classDecl,
```

Comment:
Why do we need class embedded class metadata at all?
The vtable + pointer to superclass should provide everything which is needed for method dispatch and conditional down casting.

Reply:
You're right, and that's what this function is trying to do (to emit a simple metadata record with a vtable + superclass pointer), I think the only extra leftover is `metadataBuilder.createMetadataAccessFunction();` which I'm removing in https://github.com/apple/swift/pull/68461